### PR TITLE
Add declaration DUNS validation

### PR DESCRIPTION
**Pull request by TheDoubleK**
Depends on the following changes to declaration being merged:
 - [ ] https://github.com/alphagov/digitalmarketplace-frameworks/pull/273

This adds validation to the new DUNS number field introduced with the latest declaration content changes.

### CODE REVIEW

File: tests/app/main/helpers/validation/test_g8_declaration.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    "tradingStatusOther": "Sole trader",
+    "mitigatingFactors": "It wasn't me",
+    "taxEvasion": False,
+    "primaryContact": "The boss",
+    "subcontracting": [
+        "as a prime contractor, using third parties (subcontractors) to provide all services",
+        "as part of a consortium or special purpose vehicle, using members only to provide all services"
+    ],
+    "tradingStatus": "other (please specify)",
+    "terrorism": True,
```
Comment (pcraig3) [inline tests/app/main/helpers/validation/test_g8_declaration.py]: Supplier: `"terrorism": True`

Me: :fearful:

Supplier: `"mitigatingFactors": "It wasn't me"`

Me: :smile:

File: app/main/helpers/validation.py [hunk 2]
```diff
@@ -218,7 +226,7 @@ def get_required_fields(self):
 
 
 class G8Validator(DOSValidator):
-    pass
+    number_string_fields = [('dunsNumber', 9)]
```
Comment (pcraig3) [inline app/main/helpers/validation.py]: Is DUNS number the only one we care about?  The other numbers are optional?

Comment (TheDoubleK) [inline app/main/helpers/validation.py]: Yep as far as I know DUNS number is the only thing we can specify a format for - company number, etc.  I don't think we can, as can be `N/A`.

File: tests/app/main/helpers/validation/test_g8_declaration.py (hunk 1 above)
Comment (TheDoubleK) [inline tests/app/main/helpers/validation/test_g8_declaration.py]: Yeah, we'll take them for our framework :)

**Comment by allait** [comment]
:+1:

**Pull request merged**